src/lib/encoding.ts

```diff
@@ -295,18 +297,47 @@ export function parseManifestStream(stream: Readable): EventEmitter {
       hasValidManifestVersion = true;
     }
 
-    // Index - { "index": { "path": "index.html" } }
+    // Index path - { "index": { "path": "index.html" } }
     if (
       keyPath.length === 1 &&
       keyPath[0] === 'index' &&
       currentKey === 'path'
     ) {
-      indexPath = data;
+      indexProps.path = data;
       // Resolve if the path id is already known
-      if (indexPath !== undefined && paths[indexPath] !== undefined) {
-        emitter.emit('index', { path: indexPath, id: paths[indexPath] });
-        paths = {};
+      if (
```

Comment:
Let's add an `indexProps.id === undefined` condition here.

Reply:
In the case someone sets `path` and `id`?

Reply:
Exactly. We want the `id` to win in that case (same as legacy gateway).

Reply:
Adding a `id` check will not prevent `path` to win. It will only win if `id` is parsed first.
If we want to prioritize `id` we need to ensure check if `id` exists in the case of `path` being parsed first. With this check it will only check if `id` was already parsed or not (we don't know if it exists later at this point when `path` is parsed first than `id`).

Reply:
Hmm... are you thinking about the case where `index` is after `paths`? It would prevent it otherwise, correct?

Reply:
Yes